great_expectations_cloud/agent/message_service/asyncio_rabbit_mq_client.py

```diff
@@ -228,8 +229,16 @@ def _on_channel_closed(self, channel: Channel, reason: str) -> None:
         """Callback invoked after the broker closes the channel."""
         self._close_connection()
 
-    def _build_client_parameters(self, url: str) -> pika.URLParameters:
+    def _build_client_parameters(
+        self, url: str, custom_params: dict[str, Any] | None = None
```

Comment:
```suggestion
        self, url: AmqpDsn, custom_params: dict[str, Any] | None = None
```
I don't think we need have this custom query parsing logic.
Pydantic has a `AmqpDsn` type we are alreadying using.
https://docs.pydantic.dev/1.10/usage/types/#urls

We shouldn't be casting the `AmqpDsn` into a string until we are passing it to `pika.URLParameters(str(url))`